src/components/bcros/businessDetails/index.vue

```diff
@@ -38,13 +38,13 @@ async function loadComponentData(identifier: string) {
 }
 
 // watcher required because layouts start rendering before the route is initialized
-watch(() => route.params.identifier as string, loadComponentData)
-onBeforeMount(() => {
-  // onBeforeMount required for refresh case (route will be set already so ^ watcher will not fire)
-  if (route.params.identifier) {
-    loadComponentData(route.params.identifier as string)
-  }
-})
+watch(() => route.params.identifier as string, loadComponentData, { immediate: true })
+// onBeforeMount(() => {
+//   // onBeforeMount required for refresh case (route will be set already so ^ watcher will not fire)
+//   if (route.params.identifier) {
+//     loadComponentData(route.params.identifier as string)
+//   }
+// })
```

Comment:
had an extra look at this and looks good so you can remove the commented out stuff I think